.github/workflows/docker.yml

```diff
@@ -34,26 +34,16 @@ jobs:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_TOKEN }}
 
-      # TODO: consider using https://github.com/docker/metadata-action instead of this
-      # custom magic
       - name: Calculate docker image tag
         id: set-tag
-        run: |
-          case "${GITHUB_REF}" in
-              refs/heads/develop)
-                  tag=develop
-                  ;;
-              refs/heads/master|refs/heads/main)
-                  tag=latest
-                  ;;
-              refs/tags/*)
-                  tag=${GITHUB_REF#refs/tags/}
-                  ;;
-              *)
-                  tag=${GITHUB_SHA}
-                  ;;
-          esac
-          echo "::set-output name=tag::$tag"
+        uses: docker/metadata-action@master
+        with:
+          images: ${{ secrets.DOCKER_HUB_USERNAME }}/synapse
+          tags: |
+            type=ref,event=branch
+            type=semver,pattern={{version}}
+            # set latest tag for master branch
+            type=raw,value=latest,enable=${{ github.ref == format('refs/heads/{0}', 'master') }}
```

Comment:
Is there a reason we use `format('refs/heads/{0}', 'master')` instead of the string lilteral `'refs/heads/master'`?

Reply:
Maybe this will look better?
```
type=raw,value=latest,enable=${{ github.ref == "refs/heads/master" }}
```
Thank you for your advice.

Reply:
I've tried to resolve this in 8d9b75e9, would you mind taking a look?

Reply:
Sorry, I found out that changing into string lilteral `'refs/heads/master'` won't work, I've updated this back to the format as same as [docker/metadata-action README
](https://github.com/docker/metadata-action/blob/b2391d37b4157fa4aa2e118d643f417910ff3242/README.md?plain=1#L638-L642) in 084bccd9

The testing job that didn't worked with `'refs/heads/master'` is https://github.com/henryclw/synapse-dummy-action/actions/runs/2266949247 and the testing job that worked with `github.ref == format('refs/heads/{0}', 'master')` is https://github.com/henryclw/synapse-dummy-action/actions/runs/2266951441

Reply:
Maybe it's the double quotes that's making it fail? Could you try the string literal with single quotes?

Reply:
> Maybe it's the double quotes that's making it fail? Could you try the string literal with single quotes?

Yes it works. Nice, please refer to the following links for more information:

- <https://github.com/henryclw/synapse-dummy-action/commit/170e7b322bd778c9c584888e09ae36c3b01d33a7>
- <https://github.com/henryclw/synapse-dummy-action/runs/6301952066>

Reply:
I've changed to using single quotes in bf204b9d